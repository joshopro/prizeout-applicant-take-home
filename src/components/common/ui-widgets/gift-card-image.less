@import '../../../utils/styles/mixins.less';

.gift-card-image {
	.flexbox();
	width: 100%;

	&__wrapper {
		aspect-ratio: 300/190;
		width: 100%;
		height: auto;
		position: relative;
		overflow: hidden;
		border-radius: .5rem;
	}

	&__image {
		object-fit: cover;
		object-position: center center;
		width: 100%;
		height: 100%;
		margin: 0;
		transform: scale(1.04);
		display: inline-block;
		border-radius: inherit;
	}

	&__clear-overlay {
		width: 100%;
		height: 100%;
		background-color: transparent;
		box-shadow: inset 0 0 0 2px rgba(2, 20, 49, 0.05);
		position: absolute;
		top: 0;
		left: 0;
		border-radius: inherit;
	}
}