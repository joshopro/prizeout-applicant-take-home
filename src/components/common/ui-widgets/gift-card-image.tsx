import React from 'react';
import Classnames from 'classnames';
import PropTypes from 'prop-types';

import './gift-card-image.less';

interface GiftCardImageProps {
    altText?: string;
    imgUrl: string;
}

export const GiftCardImage: React.FC<GiftCardImageProps> = ({ imgUrl, altText }): React.ReactElement => {
    const classes: string = Classnames('gift-card-image'),
        imageAltText = altText || 'Gift Card';

    return (
        <div className={classes}>
            <div className="gift-card-image__wrapper">
                <img className="gift-card-image__image" src={imgUrl} alt={imageAltText} />
                <div className="gift-card-image__clear-overlay"></div>
            </div>
        </div>
    );
};

GiftCardImage.propTypes = {
    altText: PropTypes.string,
    imgUrl: PropTypes.string.isRequired,
};
