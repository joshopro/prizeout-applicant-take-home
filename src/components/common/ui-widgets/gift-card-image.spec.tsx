import { render } from '../../../testing/test-utils';
import React from 'react';
import { GiftCardImage } from './gift-card-image';

describe('Test Gift Card Image Component', () => {
    const imageUrl = 'https://assets.prizeout.com/widget/global/generic-giftcard.png';

    test('Component matches snapshot', () => {
        const { asFragment } = render(<GiftCardImage imgUrl={imageUrl} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
