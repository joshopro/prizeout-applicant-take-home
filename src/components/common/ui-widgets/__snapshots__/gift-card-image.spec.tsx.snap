// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gift Card Image Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="gift-card-image"
  >
    <div
      class="gift-card-image__wrapper"
    >
      <img
        alt="Gift Card"
        class="gift-card-image__image"
        src="https://assets.prizeout.com/widget/global/generic-giftcard.png"
      />
      <div
        class="gift-card-image__clear-overlay"
      />
    </div>
  </div>
</DocumentFragment>
`;
